sdks/v4-sdk/src/posm.ts

```diff
@@ -0,0 +1,354 @@
+import { abi } from './utils/abi'
+
+export interface CommonOptions {
+  /**
+   * Optional data to pass to hooks
+   */
+  hookData?: string
+}
+
+export interface ModifyPositionSpecificOptions {
+  /**
+   * Indicates the ID of the position to increase liquidity for.
+   */
+  tokenId: BigintIsh
+  /**
+   * How much the pool price is allowed to move.
+   */
+  slippageTolerance: Percent
+
+  /**
+   * When the transaction expires, in epoch seconds.
+   */
+  deadline: BigintIsh
+}
+
+export interface MintSpecificOptions {
+  /**
+   * How much the pool price is allowed to move.
+   */
+  slippageTolerance: Percent
+  /**
+   * The account that should receive the minted NFT.
+   */
+  recipient: string
+
+  /**
+   * Creates pool if not initialized before mint.
+   */
+  createPool?: boolean
+
+  /**
+   * Initial price to set on the pool if creating
+   */
+  sqrtPriceX96?: BigintIsh
+}
+
+/**
+ * Options for producing the calldata to add liquidity.
+ */
+export interface CommonAddLiquidityOptions {
+  /**
+   * Whether to spend ether. If true, one of the pool tokens must be WETH, by default false
+   */
+  useNative?: NativeCurrency
+
+  /**
+   * The optional permit parameters for spending token0
+   */
+  token0Permit?: any // TODO: add permit2 permit type here
+
+  /**
+   * The optional permit parameters for spending token1
+   */
+  token1Permit?: any // TODO: add permit2 permit type here
+}
+
+/**
+ * Options for producing the calldata to exit a position.
+ */
+export interface RemoveLiquiditySpecificOptions {
+  /**
+   * The percentage of position liquidity to exit.
+   */
+  liquidityPercentage: Percent
+
+  /**
+   * Whether the NFT should be burned if the entire position is being exited, by default false.
+   */
+  burnToken?: boolean
+
+  /**
+   * The optional permit of the token ID being exited, in case the exit transaction is being sent by an account that does not own the NFT
+   */
+  permit?: NFTPermitOptions
+
+  /**
+   * Parameters to be passed on to collect
+   */
+  collectOptions: Omit<CollectSpecificOptions, 'tokenId'>
+}
+
+export interface CollectSpecificOptions {
+  /**
+   * Indicates the ID of the position to collect for.
+   */
+  tokenId: BigintIsh
+
+  /**
+   * Expected value of tokensOwed0, including as-of-yet-unaccounted-for fees/liquidity value to be burned
+   */
+  expectedCurrencyOwed0: CurrencyAmount<Currency>
+
+  /**
+   * Expected value of tokensOwed1, including as-of-yet-unaccounted-for fees/liquidity value to be burned
+   */
+  expectedCurrencyOwed1: CurrencyAmount<Currency>
+
+  /**
+   * The account that should receive the tokens.
+   */
+  recipient: string
+}
+
+export type MintOptions = CommonOptions & CommonAddLiquidityOptions & MintSpecificOptions
+export type IncreaseOptions = CommonOptions & CommonAddLiquidityOptions & ModifyPositionSpecificOptions
+
+export type DecreaseLiquidityOptions = CommonOptions & RemoveLiquiditySpecificOptions & ModifyPositionSpecificOptions
```

Comment:
BurnLiquidityOptions?

Reply:
Sorry I just meant are there going to be specific ones for burn? You could use RemoveLiquiditySpecificOptions I guess but there is no need for a liquidityPercentage to be passed in

Reply:
under the hood though if a user wants to burn we should just encode a BURN call not a decrease THEN burn